expression/builtin_cast_vec.go

```diff
@@ -1013,9 +1013,27 @@ func (b *builtinCastDecimalAsJSONSig) vecEvalJSON(input *chunk.Chunk, result *ch
 }
 
 func (b *builtinCastDurationAsJSONSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsJSONSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ReserveJSON(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		result.AppendJSON(json.CreateBinary(buf.GetDuration(i, int(types.MaxFsp)).String()))
```

Comment:
Consider using the vectorized method `buf.GoDurations`?

Reply:
When I try to implement, the testcases is error:

```
func (b *builtinCastDurationAsJSONSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
	n := input.NumRows()
	buf, err := b.bufAllocator.get(types.ETDuration, n)
	if err != nil {
		return err
	}
	defer b.bufAllocator.put(buf)
	if err = b.args[0].VecEvalDuration(b.ctx, input, buf); err != nil {
		return err
	}

	result.ReserveJSON(n)
	dur := buf.GoDurations()
	for i := 0; i < n; i++ {
		if buf.IsNull(i) {
			result.AppendNull()
			continue
		}
		result.AppendJSON(json.CreateBinary(dur[i].String()))
	}
	return nil
}

```

testcases:
```
## -0.03935824801903905, -0.03935824801903905 
## 0.5305817053265199, 0.5305817053265199 
## 0.48434396567125654, 0.48434396567125654 
## 0.6799731547387196, 0.6799731547387196 
## -0.374251286629182, -0.374251286629182 
## -0.6919388125837656, -0.6919388125837656 
## 0.09089951274035325, 0.09089951274035325 
## 0.23623798894789727, 0.23623798894789727 
## "00:00:00.353729", "353.729433ms"
----------------------------------------------------------------------
FAIL: builtin_cast_vec_test.go:76: testEvaluatorSuite.TestVectorizedBuiltinCastFunc

```

Reply:
`GetDuration` creates a new `Duration` struct each time, and you can define a `Duration` variable outside this loop to avoid this.
For example:
```
var dur types.Duration
dur.Fsp = types.MaxFsp
ds := buf.GoDurations()
for i := 0; i < n; i++ {
    ...
    dur.Duration = ds[i]
    str := dur.String()
    ...
}
```

Reply:
Thank you. I had modified.